spec.html

```diff
@@ -15329,7 +15213,7 @@ <h1>[[OwnPropertyKeys]] ( )</h1>
         1. If _trapResult_ contains any duplicate entries, throw a *TypeError* exception.
         1. Let _extensibleTarget_ be ? IsExtensible(_target_).
         1. Let _targetKeys_ be ? _target_.[[OwnPropertyKeys]]().
-        1. Assert: _targetKeys_ is a List whose elements are only String and Symbol values.
+        1. Assert: _targetKeys_ is a List whose elements are Strings or Symbols.
```

Comment:
I think it's possible to read this as meaning "whose elements are all Strings or all Symbols", which we don't intend. Here are some alternatives that don't have this problem:
- a List of elements, each of which is a String or a Symbol
- a List, each of whose elements is a String or a Symbol
- a List in which each element is a String or a Symbol
- a List whose elements are each a String or a Symbol
- a List whose every element is a String or a Symbol

Of these, I think I prefer the 3rd.

Reply:
I definitely don't think "all one or all the other" is even remotely implied or possible to infer, but any of the first three also seems fine.

Reply:
I also don't read the implication that @jmdyck does, but I am open to rewordings. Unfortunately, I don't think any of the suggestions are an improvement. In fact, I interpret the latter two suggestions to actually have the problem raised by @jmdyck.

I think something like "a List in which each element may independently be either a String or a Symbol" may be a bit clearer, but also seems unnecessarily verbose.

Reply:
We could define "property key" to be "a value of type String or Symbol" and then say "a List of property keys".

Reply:
@bakkot, there is already such a definition in <https://tc39.es/ecma262/#sec-object-type>, it just needs a `<dfn>` wrapping:

> Properties are identified using key values. A property key value is either an ECMAScript String value or a Symbol value. ...

Reply:
Sorry, yes, by "define" I meant "<dfn>".